streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -66,10 +66,10 @@ public void init(final InternalProcessorContext context) {
             valSerializer = (Serializer<V>) context.valueSerde().serializer();
         }
 
-        // if value serializers are for {@code Change} values, set the inner serializer when necessary
-        if (valSerializer instanceof ChangedSerializer &&
-                ((ChangedSerializer) valSerializer).inner() == null) {
-            ((ChangedSerializer) valSerializer).setInner(context.valueSerde().serializer());
+        // if value serializers are internal wrapping serializers that may need to be given the default serializer
+        // then pass it the default one from the context
+        if (valSerializer instanceof WrappingNullableSerializer) {
+            ((WrappingNullableSerializer) valSerializer).setIfUnset(context.valueSerde().serializer());
```

Comment:
FK Join serialization utilities face the same challenge that we previously addressed in Changed(De)Serializer. Namely, that the wrapper (de)serializer is constructed during topology building, before the default serdes are available from config, so it needs to have a mechanism for updating the inner serde at run time, once the default serde is available.

FK Join actually introduces several of these wrapper serdes, so rather than just pile on extra type-check branches, I created the concept of a (De)Serializer that wraps potentially `null` (De)Serializers. In the case that the inner serde is null, it could be set (once) to the default serde.